French translate + new feature

Kidounet opened the pull request:
Add translate for french language

Kidounet committed:
French translation

Add translation for french language

Kidounet committed:
French translation

Add translation for french language

Kidounet committed:
French translation

Add translation for french language

Kidounet committed:
Bugfix on switcher.php

organize IF bloc for better case switch
- no success if acl reject in AJAX
- non-ajax in a ELSE bloc, for no run in ajax case

Kidounet committed:
Bugfix on switcher.php

organize IF bloc for better case switch
- no success if acl reject in AJAX
- non-ajax in a ELSE bloc, for no run in ajax case

Kidounet committed:
v1.1.3 - new feature : u_switch_accounts ACL

modify ACL u_link_accounts into 2 ACL :
- u_link_accounts for add/remove link in user profile panel
- u_switch_accounts for let user switch account while no ACL for
add/remove. for example : un admin can add/remove link, but not a user.
(before, when no u_link_accounts, user can see linked accounts but can't
switch

Kidounet commented:
Hello Flerex,

I added french translate and a new feature : u_switch_accounts ACL
modify ACL u_link_accounts into 2 ACL :
- u_link_accounts for add/remove link in user profile panel
- u_switch_accounts for let user switch account while no ACL for add/remove. for example : un admin can add/remove link, but not a user. (before, when no u_link_accounts, user can see linked accounts but can't
switch)

Tested on a 3.2.3 phpbb forum.

now, user can :
- add/remove link and switch account (u_link_accounts and u_switch_accounts to YES)
- can't add/remove link but can switch account (u_link_accounts to NO and u_switch_accounts to YES) (new possibility, for let ONLY admin add/remove link)
- add/remove link but can't switch account (u_link_accounts to YES and u_switch_accounts to NO) (bad option, shall never used but possible)
- can't add/remove link and can't switch account (u_link_accounts and u_switch_accounts to NO) (like previous version when "user can link account" set to NO)

I hope it's a good feature to add for other because for us it's THE feature we're waiting since a long time ;-)

see you and thanks for this great add-on
Kidou

PS: added translate of this new ACL for all language ^_^

Flerex commented:
Hello,

First of all, thanks for your contribution. It looks pretty amazing!

Even though I can't wait to merge all your changes, I have some concerns:

What are your views on the possibility that a user's account had been created a link to it without the user's permissions and, depending on the permission configuration of the forum, they couldn't remove this link without the administration being involved? I thought about adding these two different permissions in the past, but this is something that bothers me a lot, as it is a security concern. I would like to know your opinion on the matter.

I'd also appreciate that all of the new language keys that you've added were set to the English version unless you actually know the languages. As the only languages I know are `gl` and `es` I cannot assure that the other translations are correct. Regarding the `es` and `gl` translations, they are okay except for the `ACL_U_SWITCH_ACCOUNTS` `gl` key, which is missing an “s” at the end.

And finally, regarding the changes in `switcher.php`, I would appreciate if you could remove the else statement that wraps all the non-ajax code, as this adds a new layer of unnecessary complexity to the code.

Thanks again for everything, I really appreciate everything you've done!

Kidounet commented:
Hello,

this use case is very specific, but i try to explain this :
In case of Role Playing Game (for exemple), player create an account in the forum. This account  is the account of the player, not the character. they can have access to a "pub" forum.
but each player can have 1 or more characters for playing the game. The admin is the master of the game, and can create the character, give the good access to the "in game" forum. since the players aven't password of this characters accounts, they can't change name, password, or other profile fields but when associate to theyre first account, they can switch and use this "character account" while the master (the admin) authorise that. this admin can give a character account to another player if necessary (it's our must-have option we need). it's a story of trust between player and master.

For the security, it's better : no one else admin can change links. Simple users can't change link of others. And if the admin have the admin rights, it's because he is a trusted person ;)
the founder of forum can only give "admin" rigth to someone without "founder" rigths. so he keep the absolute right in all other right (and the admin can't delete founder account without have founder rights too).

now, I changed the code from your remarks (translate in english by default and remove "else" clause for no-ajax code)

but i found another bug : in admin pannel, when select an account without linked account, and try to add new link, this accounts linked to our account and not to the selected account. I changed the acp/main_module.php and adm/style/acp_management_edit.html for correct this.

I hope it's better like that.
You can compare and try the last commit from my repo. And if all is OK for you, i'll make a new PR

Regards
Kidou
